### DISCUSSION TITLE
Possible Memory Leak?

### DISCUSSION THREAD

**Discussion by swampio** [PasaOpasen/geneticalgorithm2-6.8.7#57] [General]
This is a great resource and thank you for developing it!

I have been using it in a project of mine and I have possibly discovered a memory leak when multiple GA run()'s are ran in a loop. Please see example code below:

```python
import numpy as np
from geneticalgorithm2 import geneticalgorithm2 as ga

varbound = np.array([[0,10]]*10)

algorithm_param = {
    'max_num_iteration' : 30,
    'population_size' : 100,
    'mutation_probability' : 0.1,
    'elit_ratio' : 0.01,
    'parents_portion' : 0.3,
    'crossover_type' : 'uniform',
    'max_iteration_without_improv' : None
}

def fitness(chromosome):
    return np.sum(chromosome)

ga_model = ga(function=fitness,dimension=10,variable_type='int',variable_boundaries=varbound, algorithm_parameters=algorithm_param)


for i in range(9999):
    result = ga_model.run(no_plot=False, progress_bar_stream=None, disable_printing=True)
```

**Comment by swampio** [comment]
Apologies, this looks to have now been fixed on my end with no_plot=True

**Reply by PasaOpasen** [reply]
@swampio 

I think it's the consequence of matplotlib 

maybe plt.show() of many images makes memory leak in u r using this code in something like Spyder or Jupyter

There is no reason to use hundreds calls of `run` with showing images. U need to watch optimization process only for experimental cases or trying to find best solution of only one hard task
